packages/framework/presence/src/presenceDatastoreManager.ts

```diff
@@ -200,14 +233,95 @@ export class PresenceDatastoreManagerImpl implements PresenceDatastoreManager {
 		return entry.public;
 	}
 
-	private localUpdate(data: DatastoreMessageContent): void {
-		const content = {
+	/**
+	 * The combined contents of all queued updates. Will be undefined when no messages are queued.
+	 */
+	private queuedData: GeneralDatastoreMessageContent | undefined;
+
+	/**
+	 * Enqueues a new message to be sent. The message may be queued or may be sent immediately depending on the state of
+	 * the send timer, other messages in the queue, the configured allowed latency, etc.
+	 */
+	private enqueueMessage(
+		data: GeneralDatastoreMessageContent,
+		options: RuntimeLocalUpdateOptions,
+	): void {
+		// Merging the message with any queued messages effectively queues the message.
+		// It is OK to queue all incoming messages as long as when we send, we send the queued data.
+		this.queuedData = mergeGeneralDatastoreMessageContent(this.queuedData, data);
+
+		const { allowableUpdateLatencyMs } = options;
+		const now = Date.now();
+		const thisMessageDeadline = now + allowableUpdateLatencyMs;
+
+		if (
+			// If the deadline for this message is later than the overall send deadline, then
+			// we may be able to exit early if a timer will take care of sending it.
+			thisMessageDeadline >= this.timer.expireTime &&
+			// If the timer has not expired, we can short-circuit because the timer will fire
+			// and cover this update. In other words, queuing this will be fast enough to
+			// meet its deadline, because a timer is already scheduled to fire before its deadline.
+			!this.timer.hasExpired()
```

Comment:
Should we swap the order? It feels odd to check the expireTime on something that has expired.